src/View/Helper/FormHelper.php

```diff
@@ -2488,28 +2494,42 @@ public function getValueSources(): array
     }
 
     /**
-     * Extracts valid value sources.
+     * Validate value sources.
      *
      * @param string[] $sources A list of strings identifying a source.
-     * @return string[]
+     * @return void
+     * @throws \InvalidArgumentException If sources list contains invalid value.
      */
-    protected function extractValidValueSources(array $sources): array
+    protected function validateValueSources(array $sources): void
```

Comment:
removing old `extractValidValueSources` methods is not BC !! is ?

Reply:
No, it was added by me some days ago because sourceValues array was hardcoded in FormHelper

Reply:
If it is a protected method, then in a minor this wouldn't be too big of a deal. At least if it is very unlikely to be overwritten - which it seems in this case.

Reply:
> removing old extractValidValueSources methods is not BC !! is ?

No it's not. This method is not present in any release. It was added just few days ago to 4.next branch.

Even if it was present in 4.0, our BC policy allows changing protected methods in new minors.